uaclient/entitlements/esm.py

```diff
@@ -25,16 +25,16 @@ def disable(self, silent=False, force=False):
         if not self.can_disable(silent, force):
             return False
         series = util.get_platform_info('series')
-        repo_filename = self.repo_list_file_tmpl.format(
-            name=self.name, series=series)
-        keyring_file = os.path.join(apt.APT_KEYS_DIR, self.repo_key_file)
         entitlement_cfg = self.cfg.read_cache(
             'machine-access-%s' % self.name)['entitlement']
         access_directives = entitlement_cfg.get('directives', {})
         repo_url = access_directives.get('aptURL', self.repo_url)
         if not repo_url:
             repo_url = self.repo_url
-        apt.remove_auth_apt_repo(repo_filename, repo_url, keyring_file)
+        # We only remove the repo from the apt auth file, because ESM is a
+        # special-case: we want to be able to report on the available ESM
+        # updates even when it's disabled
+        apt.remove_repo_from_apt_auth_file(repo_url)
```

Comment:
I think we also need some additional logic here to revert the original /etc/apt/sources.list.d/ubuntu-esm-trusty.list to the content which was added in our postinst script which adds the following
```
# Written by ubuntu-advantage-tools
deb https://esm... trusty-security main 
and 
deb https://esm... trusty-updates main 
```

During ua enable esm the trusty-updates suite can be dropped if the trusty machine was not configured for 'trusty-updates'. This results in us dropping visibility to ESM packages in ESM's trusty-updates suite. We need to reconstitute that file (or at least re-add the trusty-updates suite if needed on auth disable.

Reply:
Do you think we need to resolve this in this PR, or shall I file an issue so we can cover it in a follow-up?

(This PR still gets us to a better place; without it, we'll never get any info about ESM updates in MOTD.)

Reply:
Agreed. let's deal with that separately. /me needs to get in an agile/iterative space

Reply:
I've opened #396 for this.